### PR TITLE
Fix test errors from linting issues - fixes #15

### LINKED ISSUE #15: Tests fail locally on Mac
According to the [Travis CI build history](https://travis-ci.org/sindresorhus/awesome-lint/builds), the builds are passing fine on Travis CI. Locally, I'm getting these errors when running the tests:

<details>
<pre><code>% npm test                                           /opt/awesome-lint (master)

> awesome-lint@0.1.0 test /opt/awesome-lint
> xo && ava

  rules/badge.js:19:7
  ⚠  19:7   Blocks are nested too deeply (5).                     max-depth

  index.js:10:11
  ⚠  35:19  Unexpected todo comment.                              no-warning-comments
  ✖  10:11  Unexpected chained assignment.                        no-multi-assign

  config.js:4:1
  ⚠  21:2   Unexpected todo comment.                              no-warning-comments
  ⚠  29:2   Unexpected todo comment.                              no-warning-comments
  ⚠  34:2   Unexpected todo comment.                              no-warning-comments
  ⚠  47:2   Unexpected todo comment.                              no-warning-comments
  ✖   4:1   Comments should not begin with a lowercase character  capitalized-comments
  ✖   7:2   Unnecessarily quoted property reset found.            quote-props
  ✖  63:2   Comments should not begin with a lowercase character  capitalized-comments
  ✖  64:2   Unnecessarily quoted property external found.         quote-props

  6 warnings
  5 errors
npm ERR! Test failed.  See above for more details.</code></pre>
</details>
<br>

screen shot 2017-04-05 at 9 14 07 pm

<hr>

<dl>
  <dt>Operating system</dt>
  <dd>Mac OS X El Capitan, Version 10.11.2 (15C50)</dd>
  <dt>`node` version</dt>
  <dd>v7.0.0</dd>
  <dt>`npm` version</dt>
  <dd>3.10.8</dd>
</dl>

### PR DESCRIPTION
**Pull request by cvan**
This fixes most of the errors from test failures I can reproduce locally (in issue #15), but then I encountered a bunch of issues with the paths to the test fixtures and the `execa.stdout` calls. Something seems up. Any ideas?

Stacktrace
<details>
<pre><code>% npm test                                 /opt/awesome-lint (fix-local-errors)

> awesome-lint@0.1.0 test /opt/awesome-lint
> xo && ava

  4 failed

  api › main
  /opt/awesome-lint/index.js:18

   17:   if (!readmeFile) {                                                     
   18:     return Promise.reject(new Error(`Couldn't find the file ${opts.filen…
   19:   }                                                                      

  Rejected promise returned by test

  Rejection reason:

    [Error: Couldn't find the file fixtures/main.md]

  m (index.js:18:25)
  Test.<anonymous> (test/api.js:5:16)
  step (test/api.js:16:191)
  Test.<anonymous> (test/api.js:16:99)
  Test.__dirname [as fn] (test/api.js:4:1)

  rules › badge - missing
  /opt/awesome-lint/index.js:18

   17:   if (!readmeFile) {                                                     
   18:     return Promise.reject(new Error(`Couldn't find the file ${opts.filen…
   19:   }                                                                      

  Rejected promise returned by test

  Rejection reason:

    [Error: Couldn't find the file fixtures/badge.md]

  m (index.js:18:25)
  Test.<anonymous> (test/rules.js:5:24)
  step (test/rules.js:13:191)
  Test.<anonymous> (test/rules.js:13:99)
  Test.__dirname [as fn] (test/rules.js:4:1)

  rules › badge - incorrect source
  /opt/awesome-lint/index.js:18

   17:   if (!readmeFile) {                                                     
   18:     return Promise.reject(new Error(`Couldn't find the file ${opts.filen…
   19:   }                                                                      

  Rejected promise returned by test

  Rejection reason:

    [Error: Couldn't find the file fixtures/badge2.md]

  m (index.js:18:25)
  Test.<anonymous> (test/rules.js:11:24)
  step (test/rules.js:13:191)
  Test.<anonymous> (test/rules.js:13:99)
  Test.__dirname [as fn] (test/rules.js:10:1)

  cli › [anonymous]
  /opt/awesome-lint/test/cli.js:6

   5:   const err = await t.throws(execa.stdout('../cli.js', ['fixtures/main.md…
   6:   t.regex(err.message, /Missing Awesome badge/);                          
   7: });                                                                       

  Value must match expression:

    "spawn ../cli.js ENOENT"

  Regular expression:

    /Missing Awesome badge/

  err.message
  => "spawn ../cli.js ENOENT"

  err
  => [Error: spawn ../cli.js ENOENT]

  Test.<anonymous> (test/cli.js:6:4)
  step (test/cli.js:16:191)

npm ERR! Test failed.  See above for more details.</code></pre>
</details>

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -18,20 +18,17 @@ module.exports = {
 	'fenced-code-marker': '`',
 	'file-extension': true,
 	'final-newline': true,
-	// TODO: because of https://github.com/wooorm/remark-lint/issues/77
```
Comment (SamVerschueren) [inline config.js]: Do not remove the `TODO` comments. Those are just warnings but are here for a reason.

File: config.js [hunk 2]
```diff
@@ -18,20 +18,17 @@ module.exports = {
...
 	// 'first-heading-level': 1,
 	'hard-break-spaces': true,
 	'heading-style': 'atx',
 	'link-title-style': '\'',
 	'list-item-bullet-indent': true,
 	'list-item-content-indent': true,
 	'list-item-indent': 'space',
-	// TODO: because of https://github.com/wooorm/remark-lint/issues/78
```
Comment (SamVerschueren) [inline config.js]: Do not remove

File: config.js [hunk 3]
```diff
@@ -18,20 +18,17 @@ module.exports = {
...
 	// 'list-item-spacing': true,
 	'no-auto-link-without-protocol': true,
 	'no-blockquote-without-caret': true,
 	'no-duplicate-definitions': true,
-	// TODO: because of https://github.com/wooorm/remark-lint/issues/84
```
Comment (SamVerschueren) [inline config.js]: Do not remove

File: config.js [hunk 4]
```diff
@@ -44,7 +41,6 @@ module.exports = {
 	'no-heading-punctuation': true,
 	'no-inline-padding': true,
 	'no-literal-urls': true,
-	// TODO: because of https://github.com/wooorm/remark-lint/issues/79
```
Comment (SamVerschueren) [inline config.js]: Do not remove

File: config.js [hunk 5]
```diff
@@ -71,7 +67,6 @@ module.exports = {
 	'trailing-slash': true,
 	// This rule is a good idea, but in reality it's way too slow and has too many false positives
 	// 'are-links-valid': {
-	// 	// TODO: The error message is too unclear here. Make a rule that disallows redirects
```
Comment (SamVerschueren) [inline config.js]: Do not remove

File: index.js [hunk 6]
```diff
@@ -32,10 +32,12 @@ m.report = opts => m(opts).then(file => {
 	}
 
 	messages.forEach(x => {
-		x.fatal = true; // TODO: because of https://github.com/wooorm/remark-lint/issues/65
```
Comment (SamVerschueren) [inline index.js]: Do not remove

File: rules/badge.js [hunk 7]
```diff
@@ -15,14 +15,16 @@ module.exports = (ast, file) => {
 		for (const child of node.children) {
 			if (node.depth === 1 && child.type === 'link' && child.url === badgeUrl) {
 				for (const child2 of child.children) {
-					if (child2.type === 'image') {
```
Comment (SamVerschueren) [inline rules/badge.js]: Do not make unrelated changes. The nested message is just a warning.

**Comment by SamVerschueren** [comment]
The reason the tests are failing is because the latest AVA doesn't change the working directory of the test file anymore (or something like this). Therefor you will have to make some changes in the tests.

1. [here](https://github.com/sindresorhus/awesome-lint/blob/master/test/api.js#L5) -> `test/fixtures/main.md`
2. [here](https://github.com/sindresorhus/awesome-lint/blob/master/test/cli.js#L5) -> `test/fixtures/main.md`
3. [here](https://github.com/sindresorhus/awesome-lint/blob/master/test/rules.js#L5) and [here](https://github.com/sindresorhus/awesome-lint/blob/master/test/rules.js#L11) -> prefix the paths with `test`

**Comment by cvan** [comment]
@SamVerschueren: awesome, thanks. I just pushed an update based on addressed feedback per your comments in this PR and [in this issue](https://github.com/sindresorhus/awesome-lint/issues/15#issuecomment-292072071).

**Comment by sindresorhus** [comment]
👍

**Pull request merged**